src/asl.ts

```diff
@@ -1569,32 +1563,7 @@ export class ASL {
           };
     }
 
-    if (isPromiseExpr(expr)) {
-      // if we find a promise, ensure it is wrapped in Await or returned then unwrap it
-      if (
-        expr.hasOnlyAncestors(isConditionExpr, anyOf(isReturnStmt, isAwaitExpr))
-      ) {
-        return this.eval(expr.expr);
-      }
-      throw new SynthError(
-        ErrorCodes.Integration_must_be_immediately_awaited_or_returned,
-        `Integration must be immediately awaited or returned ${nodeToString(
-          expr
-        )}`
-      );
-    } else if (isPromiseArrayExpr(expr)) {
-      // if we find a promise array, ensure it is wrapped in a Promise.all then unwrap it
-      if (
-        isArgument(expr.parent) &&
-        isCallExpr(expr.parent.parent) &&
-        isPromiseAll(expr.parent.parent)
-      ) {
-        return this.eval(expr.expr);
-      }
-      throw new SynthError(
-        ErrorCodes.Arrays_of_Integration_must_be_immediately_wrapped_in_Promise_all
-      );
```

Comment:
@thantos - i'm curious if there is another way to validate these, like using the AST representation of `Promise.all`?

Reply:
Could we add tags to our AST and include `PROMISE` as a tag?

Reply:
Wouldn't that require type information?

Reply:
That would require type information so it doesn't solve the problem. We will just rely on static validation for now and not have it inside synth.